src/compiler/binder.ts

```diff
@@ -1211,7 +1211,7 @@ namespace ts {
             bind(node.statement);
             popActiveLabel();
             if (!activeLabel.referenced && !options.allowUnusedLabels) {
-                errorOrSuggestionOnFirstToken(unusedLabelIsError(options), node, Diagnostics.Unused_label);
+                errorOrSuggestion(unusedLabelIsError(options), node.label, Diagnostics.Unused_label);

```

Comment:
this will mark function declarations as well, right?

Reply:
Right. The quick fix will skip them, but the error message will be a single span rather than several spans separated by functions.

Reply:
do you think it is worth changing it to multiple spans?

Reply:
I guess this would be ugly for a pattern like:
```ts
return { fn1, fn2, fn3, <<etc>> };
doSometing();

function fn1() { ... }
function fn2() { ... }
<<etc>>
```
that would show hundreds of lines as unreachable when only one line is.

The problem is that our current interface is:
```ts
export interface Diagnostic extends DiagnosticRelatedInformation {
    /** May store more in future. For now, this will simply be `true` to indicate when a diagnostic is an unused-identifier diagnostic. */
    reportsUnnecessary?: {};
    source?: string;
    relatedInformation?: DiagnosticRelatedInformation[];
}
export interface DiagnosticRelatedInformation {
    category: DiagnosticCategory;
    code: number;
    file: SourceFile | undefined;
    start: number | undefined;
    length: number | undefined;
    messageText: string | DiagnosticMessageChain;
}
export interface DiagnosticWithLocation extends Diagnostic {
    file: SourceFile;
    start: number;
    length: number;
}
```
So each additional span would need a new `messageText`. But it's unreachable for the same reason the first line of code was unreachable, so it doesn't feel right to add an additional span for that.

We could just add a new diagnostic for each group of unreachable statements separated by functions.

Reply:
I was thinking of multiple diagnostics, one for each contiguous span..